crypto/compactcert/builder.go

```diff
@@ -137,16 +152,12 @@ func (b *Builder) SignedWeight() uint64 {
 //
 // coinIndex works by doing a binary search on the sigs array.
 func (b *Builder) coinIndex(coinWeight uint64) (uint64, error) {
-	if !b.sigsHasValidL {
-		return 0, fmt.Errorf("coinIndex: need valid L values")
-	}
-
 	lo := uint64(0)
 	hi := uint64(len(b.sigs))
 
 again:
 	if lo >= hi {
-		return 0, fmt.Errorf("coinIndex: lo %d >= hi %d", lo, hi)
+		return 0, fmt.Errorf("%w: lo %d >= hi %d", ErrInternalCoinIndexError, lo, hi)
```

Comment:
This should never happen in the code path. It is expected to have: 
```
b.sigs[0].L = 0 // by construction
b.sigs[last].L + b.sigs[last].Weight == signedWeight  // b.sigs[last].L + b.sigs[last].Weight == signedWeight and coinWeight in [0, signedWeight) 
```
In case it happens, the error should say something like: 
coinIndex unexpected behavior: lo = %d hi = %d coinWeight = %d